### ISSUE TITLE
[Feature request]: Update System Requirements for Windows XP

### ISSUE THREAD

**Issue by dftf-stu** [project64/project64#2253]
### Request box

**GitHub**
- The "System Requirements" section, on the homepage (https://github.com/project64/project64) of this project, needs updating regarding _Windows XP_ support
- The "Operating System" section should be updated to state what the minimum supported OS now is (for the current, stable version) and for _XP_ to state what the final version is that will run

**Project64 Website**
- On the "Download Project64" page (https://www.pj64-emu.com/public-releases) each download should have the "For Windows" text amended to clarify which OS they run on (e.g. "For Windows 7 or later" and "last version for Windows XP")

It would be great to see whoever manages these two areas update them

### Reason box

In my testing, version 2.3.2.202 (released 4 Feb 2019) is the most-recent version I could get working under Windows XP Home SP3.

However, the current _System Requirements_ section on the GitHub homepage implies that the current, stable version should still work on _Windows XP_, as it states that as the minimum with no-further qualifications:

![SystemSpecs](https://user-images.githubusercontent.com/83965343/183243917-0f2f6ab1-99d2-42fa-8b29-910cda1889d2.png)

The current stable version, v3.0.1, does not work on _Windows XP_.

*          *          *

Similarly, if you go to the official website for Project64, the "Download Project64" page also does not state for each download which OS they support:

![SystemRequirements_MainSite](https://user-images.githubusercontent.com/83965343/180004723-49c2c5a7-004c-468f-a226-845005d4a884.png)

### Examples

**_DuckStation_** makes-clear on their GitHub homepage (https://github.com/stenzek/duckstation) that:
- Nothing below Windows 10 is supported
- Also mentions any extras (install latest Visual C++; enable TLS 1.2 on _Windows 7_) that the user must do

**_PCSX2_** is even-clearer (https://github.com/PCSX2/pcsx2), with a table-format showing:
- Windows 8.1 or newer is required, and only 64-bit installs of _Windows_ are supported
- That the CPU must support the SSE4.1 instruction
- And how-much RAM is required, along with GPU RAM and library-support levels

**Comment by dftf-stu** [comment]
Worth also noting, the v2.4.0.1868 release (and newer milestones) also does not work on **Windows Vista**

Unlike on _Windows XP_, the main window does actually appear...

![VISTA_01](https://user-images.githubusercontent.com/83965343/180043852-3bd3fa1c-1f60-4498-9f1e-256fe933e8b9.png)

...but the moment you go to load a ROM, you then get this error:

![VISTA_02](https://user-images.githubusercontent.com/83965343/180043906-e66c9f4d-44cc-4b75-8a57-924daf722e1d.png)


According to [this blog post](https://devblogs.microsoft.com/cppblog/windows-sdk-v7-0v7-0a-incompatibility-workaround/) by Microsoft, this is because from Windows 7 onwards, some system-calls and APIs were moved out-of "psapi.dll" and into "kernel32.dll".  So it would seem on _Vista_, _Project64_ does not look in the older DLL, only in KERNEL32.DLL.

**Comment by DerekTurtleRoe** [comment]
In my opinion what we should really do is just support the latest version of Windows 10, since all others are either already EOL or are very close to being EOL.

Nobody is updating drivers, software, or firmware to support anything besides Windows 7 and above, and support for Windows 7, Windows 8, and Windows 8.1 is already very scarce.

**Comment by Squall-Leonhart** [comment]
its because the currently used Toolchain is not the XP compatible Toolchain, that can be easily fixed.

**Comment by DerekTurtleRoe** [comment]
Yeah, like I said in the other thread, I'm sure it can be easily fixed.

**Comment by Squall-Leonhart** [comment]
Need to use EnumProcessModules instead of K32EnumProcessModules too, its not a significant  change and the behaviour is the same afaik

**Comment by dftf-stu** [comment]
I agree with @DerekTurtleRoe: if 2.3.2 is the last version supported on XP and Vista, just update the README on the homepage here to state that is the case.  And then decide when you will drop Windows 7 and 8.1 support.  For example:

**Operating System**
• Windows 10 or later recommended
• Windows 7 and 8.1: the latest Stable and Dev releases work, but our next major Dev milestone (5.0.x) will drop support
• Still using XP or Vista?  Version 2.3.2 is the last to support your OS

**Comment by dftf-stu** [comment]
So I notice @Squall-Leonhart has updated the System Requirements (#2262), which is great to see.

BUT... any chance that "Windows XP and later" could also be changed to "Windows 7 and later", given that the most-recent versions of _Project64_ don't work on XP (unless you disable the Discord support) or Vista (due to [API changes](https://devblogs.microsoft.com/cppblog/windows-sdk-v7-0v7-0a-incompatibility-workaround/) that were introduced in Windows 7)?

Until various members of the development team have discussed whether it's worth making the current versions work in those older versions (which I'd argue likely not, given most stat websites put XP use on 0.4% or less, and Vista on 0.1% or less), then I'd simply argue this should be amended until such time any decision is made to avoid confusion before then.

Thanks

**Comment by Squall-Leonhart** [comment]
the status of XP and  Vista support are not confirmed either way as yet.

**Comment by dftf-stu** [comment]
Hi @flagrama , @DerekTurtleRoe , @project64 

So, following on from my comments in ticket 2382, I've just done some testing of a mixture of recent Project64 versions inside a cleanly-installed Windows XP Home Edition with Service Pack 3.  This was inside a VMware Workstation Player VM, which had 3D-passthrough enabled (as I no-longer have any devices that run XP natively, as I guess most don't thesedays!)

For the testing I simply did the following:
- For the ZIP versions of Project64, I extracted them into their own folders, selected my ROM folder, and choose "Project64 Video" as the plugin; for the 2.3.2-202 release, which is only in EXE form, I used the "portable mode" option to extract its files
- I used "Super Mario 64" (North-American version, MD5: 49FBF9C744C458405F685D42B762A0BE) as my test
- I made no changes to any of the settings in any Project64 instance, except disabling Discord where necessary (more below), and after testing with the "Project64 Video", changed the default to "Jabo's Direct3D", then exited and relaunched the app

SUMMARY
- The 2.3.2-202 release is the last one that fully-works "out-of-the-box" on Windows XP SP3
- The 4.0.0-6151, 3.0.1-5670, 3.0.0-5648 and 2.4.0-1868 releases all require Discord to be disabled before they will run
- The 4.0.0-6151, 3.0.1-5670, 3.0.0-5648 and 2.4.0-1868 releases all display an error before the "Configuration" dialog appears
- The 3.0.1-5670, 3.0.0-5648, 2.4.0-1868 and 2.3.2-202 releases could all load the ROM using both video-plugins (PJ64 or JD3D)
- The 4.0.0-6151 release crashes whenever launching the ROM; it only works if "Always use interpreter core" is enabled

SCREENSHOTS

![Screenshot 2023-08-16 195235](https://github.com/project64/project64/assets/83965343/72adc32b-9cff-43ba-a8d5-173df4bb8e86)
**Above:** users must edit the "Project64.cfg" file and add the "Enable Discord RPC=0" line manually before the app will even start on Windows XP.  If the app shipped with this off-by-default, users would be able to at-least start the app...

![Screenshot 2023-08-16 195307](https://github.com/project64/project64/assets/83965343/6b62f654-7e3a-4446-b1d1-f53b0d088c37)
**Above:** when you go to Options > Configuration, expect to see this error every-single-time in every version except the 2.3.2-202 release.  After the user clicks "OK", the "Configuration" dialog does then load up, and can be used

![Screenshot 2023-08-16 195535](https://github.com/project64/project64/assets/83965343/79480671-1c7e-4fe0-9128-3a2ba3b6de1c)
**Above:** I was able to run Super Mario 64 (NA version) on all versions of Project64, on both the "Project64 Video" and "Jabo's Direct3D" video-plugins, EXCEPT for 4.0.0-6151, where the app simply stops-running after a few seconds of a blank-screen.  However, if I enable "Always use interpreter core" in that version, then the game does run under either plug-in

**Comment by dftf-stu** [comment]
So... just out of interest: I notice on the main-page here, the "[Supported requirements](https://github.com/project64/project64/?tab=readme-ov-file#supported-requirements)" section has been updated to say "64-bit Windows 10 and 11".

So... given it now seems _someone_ has finally taken a decision to drop _XP_ and _Vista_ (along with _7_ and _8.x_), might I suggest just adding links to the final versions that are known to work on each OS, and then close this ticket (and similar ones)? 🤔

For example, it could look something like this:

### **DOWNLOAD**

- Windows 10 (64-bit only) or 11 users: grab the latest [stable] here
You can also try a [nightly] version, but these break often and aren't guaranteed to always work.  You might have to download a few versions until you find one that runs on your system -- be sure to unzip each download into a separate folder.

> ⚠Windows 11 on ARM, Windows 10 on ARM and Windows RT devices are NOT supported at-all.  Do NOT log any tickets on here for issues on those platforms; they will be immediately closed.  Windows 10 (32-bit) and older are also unsupported.  We recommend all Windows 10 (64-bit) users to be running the latest 22H2 release.

**Using an older version of Windows?**

> ℹ You're free to use an older version, but we can't guarantee all games will work, and we are no-longer updating them.  We will also not accept any tickets to be raised on here -- tickets are accepted for Windows 10 (64-bit) or 11 systems only.

- Windows 10 (32-bit), 8.x and 7 users: use the [3.0.1 stable] release
- Windows Vista and XP users: use the [2.3.2 stable] release

We recommend users check the latest Service Pack for their OS (7 SP1; Vista SP2; XP SP3) is installed.
